![DOC](documentacion.jpg)


In [24]:
#Cargamos las librerias necesarias 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import os
# Configuro pandas para mostrar todas las columnas
pd.set_option('display.max_columns', None)

In [25]:
# Directorio donde se encuentran los archivos XLSX
directory = r'C:\Users\sosaj\OneDrive\Escritorio\Andata-\Mondelez\revisar'

# Lista de archivos xlsx en el directorio
files = [f for f in os.listdir(directory) if f.endswith('.xlsx')]

# Leo cada archivo y lo almaceno en un DF con nombre dinamico
record_counts = {}
for i, file in enumerate(files):
    df_name = f'df{i + 1}'
    file_path = os.path.join(directory, file)
    globals()[df_name] = pd.read_excel(file_path)
    record_counts[df_name] = globals()[df_name].shape[0]  # numero de filas
    print(f'{df_name} cargado con éxito, número de registros: {record_counts[df_name]}')

# Verifico las columnas en cada DF
for df_name in ['df1', 'df2', 'df3', 'df4', 'df5', 'df6', 'df7']:
    df = globals().get(df_name)
    if df is not None:
        print(f'{df_name} columnas: {df.columns.tolist()}')

# Selecciono solo las columnas de interes en cada df
df1 = df1[['identificador_pdv', 'Canal', 'Provincia']]
df2 = df2[['Identificador_pdv_censo', 'Canal', 'Provincia']]
df3 = df3[['Identificador', 'Canal_Unificado', 'Provincia']]
df4 = df4[['Identificador_pdv', 'Canal', 'Provincia']]
df5 = df5[['identificador_pdv (censo)', 'Canal', 'Provincia']]
df6 = df6[['identificador_pdv', 'Canal', 'Provincia']]
df7 = df7[['identificador_pdv', 'Canal', 'Provincia']]

# Renombro las columnas para que todas tengan los mismos nombres
df2.rename(columns={'Identificador_pdv_censo': 'identificador_pdv'}, inplace=True)
df3.rename(columns={'Identificador': 'identificador_pdv', 'Canal_Unificado': 'Canal'}, inplace=True)
df5.rename(columns={'identificador_pdv (censo)': 'identificador_pdv'}, inplace=True)

# Verifico el nombre de la columna en df4
df4.rename(columns={'Identificador_pdv': 'identificador_pdv'}, inplace=True)

# Verificar NA en cada DataFrame antes de concatenar
for df_name in ['df1', 'df2', 'df3', 'df4', 'df5', 'df6', 'df7']:
    df = globals().get(df_name)
    na_count = df['identificador_pdv'].isna().sum()
    print(f'NA en {df_name} antes de concatenar: {na_count}')

# Concatenar todos los Dfs en uno solo
df_final = pd.concat([df1, df2, df3, df4, df5, df6, df7], ignore_index=True)

# Verifico NA en el df final
na_rows = df_final[df_final['identificador_pdv'].isna()]
print(f'Número de filas con NA en "identificador_pdv" después de la concatenación: {len(na_rows)}')
print(na_rows.head())

# Elimio columnas extra 
df_final = df_final[['identificador_pdv', 'Canal', 'Provincia']]

df1 cargado con éxito, número de registros: 9577
df2 cargado con éxito, número de registros: 1898
df3 cargado con éxito, número de registros: 828
df4 cargado con éxito, número de registros: 442
df5 cargado con éxito, número de registros: 6022
df6 cargado con éxito, número de registros: 1394
df7 cargado con éxito, número de registros: 2225
df1 columnas: ['Nombre del Censista', 'identificador_pdv', 'Nombre de Alta', 'Dirección (Generada por el Censita)', 'Fecha', 'Provincia', 'Departamento', 'Localidad', 'Coordenadas', 'Coordenadas fin', 'Origen del Registro', 'Canal', 'Razón Social', 'CUIT', 'Papeles?', 'Cantidad de Góndolas(Autoservicio)', 'Tiene Verdulería(Autoservicio)', 'Tiene Fiambrería(Autoservicio)', 'Heladeras con bebidas(Autoservicio)', 'Cantidad de Exhibidores de primeras marcas(Almacen)', 'Tiene surtido de vinos(Almacen)', 'Tiene Heladera de lácteos(Almacen)', 'Tiene gondola(Maxi Kiosco)', 'si tenga gondola, cuan llena esta?(Maxi Kiosco)', 'Tiene exhibidor de chicles/caramelo

In [27]:
# Obtener el diccionario de recuentos de registros del entorno global
record_counts = globals().get('record_counts', {})

# Verificar si record_counts contiene datos
if not record_counts:
    print("El diccionario 'record_counts' está vacío o no se encontró.")
else:
    # Sumar las cantidades de registros
    total_registros = sum(record_counts.values())
    print(f'Número total de registros en todos los DataFrames: {total_registros}')


print(df_final.shape)

Número total de registros en todos los DataFrames: 22386
(22386, 3)


In [28]:
df_final

identificador_pdv                  Canal            Provincia
0            ANZ-541694                Almacén  Santiago del Estero
1            BRR-835378            Maxi Kiosco  Santiago del Estero
2            CHI-157628            Maxi Kiosco  Santiago del Estero
3            DEJ-431859              Rotisería  Santiago del Estero
4            JA -001052             Golsinería  Santiago del Estero
...                 ...                    ...                  ...
22381        YRM-624933  Farmacia / Perfumería           Entre Ríos
22382        YUA-939181               Vinoteca           Entre Ríos
22383        ZAI-174258                  Hotel           Entre Ríos
22384        ZEM-972541                   Club           Entre Ríos
22385        ZOA-566651              Retaurant           Entre Ríos

[22386 rows x 3 columns]

In [29]:
df_puntos_de_interes  = pd.read_excel("acumulado_10.xlsx")

veremos que todos los identificadores de df_final esten en df_puntos_de_interes

In [30]:

# Obtener los identificadores únicos de ambos DataFrames
identificadores_final = set(df_final['identificador_pdv'].dropna())
identificadores_puntos_de_interes = set(df_puntos_de_interes['Identificador'].dropna())

# Verificar qué identificadores de df_final están en df_puntos_de_interes
coincidencias = identificadores_final.intersection(identificadores_puntos_de_interes)
total_identificadores_final = len(identificadores_final)
total_coincidencias = len(coincidencias)

# Calcular el porcentaje de coincidencia
porcentaje_coincidencia = (total_coincidencias / total_identificadores_final) * 100 if total_identificadores_final > 0 else 0

# Imprimir los resultados
print(f'Número total de identificadores en df_final: {total_identificadores_final}')
print(f'Número de identificadores coincidentes en df_puntos_de_interes: {total_coincidencias}')
print(f'Porcentaje de coincidencia: {porcentaje_coincidencia:.2f}%')

# Identificar los identificadores que están en df_final pero no en df_puntos_de_interes
no_coincidentes = identificadores_final.difference(identificadores_puntos_de_interes)
print(f'Identificadores en df_final pero no en df_puntos_de_interes: {no_coincidentes}')


Número total de identificadores en df_final: 21819
Número de identificadores coincidentes en df_puntos_de_interes: 21814
Porcentaje de coincidencia: 99.98%
Identificadores en df_final pero no en df_puntos_de_interes: {'K6-089634', 'A -966566', 'ALO-530575', 'CAA-448853', 'LA-191259'}


Como podemos observar tenemos para estas bases entregadas una coicidencia del 99.98% de coincidencias por lo que estos datos no presentan errores de exportacion en por comillas. 